Date: Thu, 14 Jun 2016 14:32:00 +0530
Subject: [PATCH] frameworks/base: PowerManagerService reboot recovery fix
Rebased on TeamButter's patches
---
diff --git a/services/java/com/android/server/power/PowerManagerService.java b/services/java/com/android/server/power/PowerManagerService.java
index 56a4096..8fe9ad7 100644
--- a/services/java/com/android/server/power/PowerManagerService.java
+++ b/services/java/com/android/server/power/PowerManagerService.java
@@ -2676,12 +2676,15 @@ public final class PowerManagerService extends IPowerManager.Stub
             // This preparation can take more than 20 seconds if
             // there's a very large update package, so lengthen the
             // timeout.
-            SystemProperties.set("ctl.start", "pre-recovery");
+            
+            // For logan2g: "ctl.start, pre-recovery" mode does not work,
+            // so use "sys.powerctl, reboot, reason=recovery" to boot into recovery.
+ 
             duration = 120 * 1000L;
         } else {
-            SystemProperties.set("sys.powerctl", "reboot," + reason);
             duration = 20 * 1000L;
         }
+        SystemProperties.set("sys.powerctl", "reboot," + reason);
         try {
             Thread.sleep(duration);
         } catch (InterruptedException e) {
